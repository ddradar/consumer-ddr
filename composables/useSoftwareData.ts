import type { ParsedContent } from '@nuxt/content/dist/runtime/types'
import type { Ref } from 'vue'

import type { Software } from '~~/src/content'

export type SoftwareParsedContent = ParsedContent & Software

export default async function (slug: string) {
  const { data } = await useAsyncData(`/software/${slug}`, () =>
    queryContent<SoftwareParsedContent>(slug)
      .where({ _type: 'markdown', slug })
      .findOne()
  )

  return { software: data as Ref<SoftwareParsedContent> }
}
