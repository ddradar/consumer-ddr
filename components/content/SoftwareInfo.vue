<template>
  <table class="table">
    <tbody>
      <tr>
        <th>発売日</th>
        <td>{{ software.launched }}</td>
      </tr>
      <tr>
        <th>機種</th>
        <td>{{ software.platform }}</td>
      </tr>
      <tr>
        <th>リージョン</th>
        <td>{{ software.region }}</td>
      </tr>
    </tbody>
  </table>
</template>

<script lang="ts" setup>
import useSoftwareData from '~~/composables/useSoftwareData'

interface Props {
  series: string
}

const prop = defineProps<Props>()

const { software } = await useSoftwareData(prop.series)
</script>
