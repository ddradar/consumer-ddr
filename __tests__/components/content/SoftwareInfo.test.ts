import { describe, expect, test, vi } from 'vitest'
import { ref } from 'vue'

import SoftwareInfo from '~~/components/content/SoftwareInfo.vue'
import useSoftwareData from '~~/composables/useSoftwareData'

import { mockSoftware, mountAsync } from '../../test-utils'

vi.mock('~~/composables/useSoftwareData')

describe('components/content/SoftwareInfo.vue', () => {
  describe('snapshot test', () => {
    test('renders correctly', async () => {
      // Arrange
      vi.mocked(useSoftwareData).mockResolvedValue({
        software: ref(mockSoftware)
      })

      // Act
      const props = { series: mockSoftware.slug } as any
      const wrapper = await mountAsync(SoftwareInfo, { props })

      // Assert
      expect(wrapper.element).toMatchSnapshot()
    })
  })
})
