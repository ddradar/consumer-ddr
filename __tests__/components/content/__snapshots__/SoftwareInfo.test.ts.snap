// Vitest Snapshot v1

exports[`components/content/SoftwareInfo.vue > snapshot test > renders correctly 1`] = `
<table
  class="table"
>
  <tbody>
    <tr>
      <th>
        発売日
      </th>
      <td>
        1999-04-10
      </td>
    </tr>
    <tr>
      <th>
        機種
      </th>
      <td>
        Play Station
      </td>
    </tr>
    <tr>
      <th>
        リージョン
      </th>
      <td>
        JP
      </td>
    </tr>
  </tbody>
</table>
`;
